client/src/pages/pot-instance/components/SimpleTimerView/SimpleTimerView.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable no-unused-vars */
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import StackedProgressBar from 'components/StackedProgressBar';
+import { PotContent, ProgressData } from 'pages/pot-instance/models';
+import getRGBPerCategory from 'utils/colors';
+
+interface Props {
+  potContent: PotContent[];
+  setPotContent: Dispatch<SetStateAction<PotContent[]>>;
+}
+
+interface Data extends PotContent {
+  color: string;
+}
+
+const SimpleTimerView = ({ potContent, setPotContent }: Props) => {
+  console.log(setPotContent);
+  const [data, setData] = useState<Map<string, Data>>(new Map());
+
+  const [outputData, setOutData] = useState<ProgressData[]>([]);
+  useEffect(() => {
+    potContent.forEach((potItem) => {
+      data.set(potItem.id, {
+        ...potItem,
+        color: data.get(potItem.id)?.color || getRGBPerCategory()
+      });
+    });
+    // TODO: Clean up renmoved items
+    setData(new Map(data));
+  }, [potContent, setData]);
+
+  useEffect(() => {
+    const temp = Array.from(data.values()).map((d) => ({
+      id: d.id,
+      name: d.name,
+      color: d.color,
+      value: (d.timeLeft / d.cookTime) * 100
+    }));
+
+    setOutData(temp);
+  }, [data, setOutData]);
```

Comment:
Whenever data gets updated, this useEffect runs to create a new Array using the information and outputs it to OutData? Would it work if this useEffect is removed and function moved to the other useEffect with the data / setOutData in the dependencies?

Reply:
Yea I was trying to see if I could have avoided the above useEffect, and directly update the potContent with color.
Might still be an option, just some more digging into.